openedx/core/djangoapps/content/learning_sequences/api/processors/enrollment_track_partition_groups.py

```diff
@@ -0,0 +1,61 @@
+# lint-amnesty, pylint: disable=missing-module-docstring
+import logging
+
+from common.lib.xmodule.xmodule.partitions.enrollment_track_partition_generator import (
+    create_enrollment_track_partition_with_course_id
+)
+from common.lib.xmodule.xmodule.partitions.partitions import (
+    ENROLLMENT_TRACK_PARTITION_ID,
+)
+from common.lib.xmodule.xmodule.partitions.partitions_service import get_user_partition_groups
+
+from .base import OutlineProcessor
+
+log = logging.getLogger(__name__)
+
+
+class EnrollmentTrackPartitionGroupsOutlineProcessor(OutlineProcessor):
+    """
+    Processor responsible for applying all enrollment track user partition group to the outline.
+
+    Confining the processor to only EnrollmentTrack user partition is a significant limitation.
+    Nonetheless, it is a step towards the goal of supporting
+    all partition schemes in the future
+    """
+    def __init__(self, course_key, user, at_time):
+        super().__init__(course_key, user, at_time)
+        self.user_groups = []
+        self.enrollment_track_groups = {}
+
+    def load_data(self):
+        user_partition = create_enrollment_track_partition_with_course_id(self.course_key)
+        self.enrollment_track_groups = get_user_partition_groups(self.course_key, [user_partition], self.user)
```

Comment:
Since we're not reaching for a general solution in this PR, I think you could further simplify this to just track the one enrollment track group the user has (e.g. `self.enrollment_track_group`, instead of loading a dict that will only ever have one key and one value.

Though actually, do people always have an enrollment track associated with them? Is there any time there would be no group at all? If that's the reason the code is structured like this, please add a comment to that effect.

Reply:
I do think user's enrollment will always have an enrollment mode. I structured my code this way because the code signature is just this way. 
For enrollment track partition, the learner should only have 1 enrollment mode active for the course. I can simplify the code